pkg/dhclient/dhcp6.go

```diff
@@ -36,34 +33,39 @@ func (p *Packet6) Link() netlink.Link {
 
 // Configure configures interface using this packet.
 func (p *Packet6) Configure() error {
-	return Configure6(p.iface, p.p, p.iana)
+	return Configure6(p.iface, p.p)
 }
 
 func (p *Packet6) String() string {
-	return fmt.Sprintf("IPv6 DHCP Lease IP %s", p.Lease().IP)
+	return fmt.Sprintf("IPv6 DHCP Lease IP %s", p.Lease().IPv6Addr)
 }
 
 // Lease returns lease information assigned.
-func (p *Packet6) Lease() *dhcp6opts.IAAddr {
-	// TODO: Can a DHCPv6 server return multiple IAAddrs for one IANA?
-	// There certainly doesn't seem to be a way to request multiple other
-	// than requesting multiple IANAs.
-	iaAddrs, err := dhcp6opts.GetIAAddr(p.iana.Options)
-	if err != nil || len(iaAddrs) == 0 {
+func (p *Packet6) Lease() *dhcpv6.OptIAAddress {
+	// TODO(chrisko): Reform dhcpv6 option handling to be like dhcpv4.
+	ianaOpt := p.p.GetOneOption(dhcpv6.OptionIANA)
```

Comment:
Is there a possibility that this use of GetOneOption may return an option that comes from a different result?

I totally don't understand how dhcp works, but suppose there are multiple IANAopts, multiple iaaddrs, and multiple DNS server sets. Could we end up mistakingly picking one iana opt meant for another iaaddr and a different set of DNS servers?

Reply:
There is, but only if you sent more than one IANA request. If you only sent one, it'd be a malformed packet to get a different one back.